src/vector/_lib.py

```diff
@@ -11,6 +11,16 @@
 
 class SympyLib:
     # functions modified specifically for sympy
+
+    # should NOT be used as a replacement for np.where as it works specifically for the
+    # case of vector/_compute/spatial/eta.py. `where` returns the second argument and
+    # ignores the third (first argument is a boolean condition) because the purpose of
+    # this function is to handle exceptional values — we know that the "normal" values
+    # are in the second argument and the "exceptional" ones are in the third argument.
+    # TODO: remove once https://github.com/cupy/cupy/issues/9143 is fixed.
+    def where(self, val1: sympy.Expr, val2: sympy.Expr, val3: sympy.Expr) -> sympy.Expr:
```

Comment:
This might be considered a hack/quick fix for the SymPy backend (as explained in the comment above). The only issue with this PR is that we will have to manually check that `where` is not used as a regular `np.where` in compute functions.